common/include/pcl/impl/point_types.hpp

```diff
@@ -2417,6 +2450,31 @@ namespace traits
   template <typename PointT>
   using HasNoColor = std::enable_if_t<!has_color_v<PointT>, bool>;
 
+  /** Metafunction to check if a given point type has either rgb or rgba field. */
+  template <typename PointT>
+  struct has_rgb : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
+                                                              pcl::fields::rgba> >
+  { };
+
+  template <typename PointT>
+  constexpr auto has_rgb_v = has_rgb<PointT>::value;
+
+  template <typename PointT>
+  using HasRGB = std::enable_if_t<has_rgb_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoRGB = std::enable_if_t<!has_rgb_v<PointT>, bool>;
+
+  /** Metafunction to check if a given point type has any color (e.g. rgb, rgba, hsv, lab) field. */
+  template <typename PointT>
+  constexpr auto has_any_color_v = has_rgb_v<PointT> || has_HSV_v<PointT> || has_Lab_v<PointT> || has_intensity_v<PointT>;
+
+  template <typename PointT>
+  using HasAnyColor = std::enable_if_t<has_any_color_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoAnyColor = std::enable_if_t<!has_any_color_v<PointT>, bool>;
```

Comment:
HasNoColor? This does mean a behavior change for `HasNoColor`, but HasNoAnyColor doesn't feel right.

Any better ideas?

Reply:
PS: You need to change the current HasNoColor to use has_rgb instead

Reply:
There is no `HasNoColor` used in current PCL source codes. But not sure whether this behavior change would be addressed in this version.

Could we just remove `HasNoAnyColor` now and defer the behavior change of `HasNoColor` until the complete deprecation in v1.15?

Reply:
please add a corresponding comment for the same as a deprecation message

Reply:
I'd do exactly that! :smile: